core/src/main/java/org/apache/iceberg/metrics/CommitReport.java

```diff
@@ -19,12 +19,15 @@
 package org.apache.iceberg.metrics;
 
 import java.util.Map;
+import org.apache.iceberg.TableOperations;
 import org.immutables.value.Value;
 
 /** A commit report that contains all relevant information from a Snapshot. */
 @Value.Immutable
 public interface CommitReport extends MetricsReport {
 
+  TableOperations tableOperations();
```

Comment:
that shouldn't be added to a `CommitReport`. why is this required here?

Reply:
I don't think there should be any coupling between `TableOperations` and any subtype of `MetricsReport`. The purpose of a `MetricsReport` is to only carry final results from a scan/commit

Reply:
Curious to know if we can have a new interface for such type of operations. Thoughts?